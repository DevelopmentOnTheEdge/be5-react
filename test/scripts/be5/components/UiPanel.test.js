import React from 'react';
import renderer from 'react-test-renderer';
import '../../../../src/scripts/be5/be5init'
import {TestProvider} from "../testUtils";
import {Document} from "../../../../src/scripts/be5/containers/Document";
import be5 from "../../../../src/scripts/be5/be5";


test('snapshot', () => {
  const component = renderer.create(
    <TestProvider>
      <Document frontendParams={{documentName: "MainDocument"}}/>
    </TestProvider>
  );

  be5.url.process('MainDocument', "#!uiPanel" );
  expect(component.toJSON()).toMatchSnapshot();
});
