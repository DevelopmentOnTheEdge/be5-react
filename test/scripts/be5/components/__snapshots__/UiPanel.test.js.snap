// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="ui-panel row"
>
  <div
    className="col-md-12"
  >
    <h1>
      Core
    </h1>
  </div>
  <div
    className="col-md-4"
  >
    <h3>
      documents
    </h3>
    <div>
      <span
        className="badge badge-primary"
      >
        errorPane
      </span>
       - 
      ErrorPane
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        form
      </span>
       - 
      HorizontalForm
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        formTable
      </span>
       - 
      FormTable
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        inlineMiniForm
      </span>
       - 
      InlineMiniForm
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        modalForm
      </span>
       - 
      ModalForm
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        operationResult
      </span>
       - 
      FinishedResult
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        queryBuilder
      </span>
       - 
      QueryBuilder
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        rTable
      </span>
       - 
      ReactTable
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        static
      </span>
       - 
      StaticPage
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        submitOnChange
      </span>
       - 
      SubmitOnChangeForm
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        table
      </span>
       - 
      Table
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        tableForm
      </span>
       - 
      TableForm
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        tableFormRow
      </span>
       - 
      TableFormRow
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        uiPanel
      </span>
       - 
      UiPanel
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        verticalForm
      </span>
       - 
      Form
    </div>
  </div>
  <div
    className="col-md-8"
  >
    <h3>
      routes
    </h3>
    <div>
      <span
        className="badge badge-primary"
      >
        categories
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        entity
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        form
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        entity
      </span>
      <span>
        , 
        query
      </span>
      <span>
        , 
        operation
      </span>
      <span>
        , 
        operationParams
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        loading
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        page
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        login
      </span>
      (
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        logout
      </span>
      (
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        queryBuilder
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        params
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        static
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        page
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        table
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        entity
      </span>
      <span>
        , 
        query
      </span>
      <span>
        , 
        params
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        text
      </span>
      (
      <span>
        documentName
      </span>
      <span>
        , 
        text
      </span>
      )
    </div>
    <div>
      <span
        className="badge badge-primary"
      >
        uiPanel
      </span>
      (
      <span>
        documentName
      </span>
      )
    </div>
  </div>
</div>
`;
